src/macaron/slsa_analyzer/analyzer.py

```diff
@@ -942,6 +988,99 @@ def create_analyze_ctx(self, component: Component) -> AnalyzeContext:
 
         return analyze_ctx
 
+    def get_artifact_hash(
+        self,
+        purl: PackageURL,
+        cached_artifacts: list[str] | None,
+        hash_algorithm: Any,
+        package_registries_info: list[PackageRegistryInfo],
+    ) -> str | None:
+        """Get the hash of the artifact found from the passed PURL using local or remote files.
```

Comment:
Could we add more description on the behaviors of this function regarding the local or remote files. For example
- When a remote files are fetched ? Are they stored persistently in memory ?
- The function returns early if a local artifact is discovered.